### PR TITLE
[POT] set shape in FBC if undefined dimension in model

### PR DESCRIPTION
**Pull request by IndiraSalyahova**
### Tickets:
 - 73787

### CODE REVIEW

File: tools/pot/openvino/tools/pot/algorithms/quantization/fast_bias_correction/algorithm.py [hunk 1]
```diff
@@ -202,3 +203,16 @@ def get_channel_axis(self, node):
...
         return self._channel_axis[node]
+
+    def get_inputs_shape(self, nodes_with_bias):
+        sampler = create_sampler(self._engine, 1, False, 0)
+        calculate_input_shape = {}
+        for op_node in nodes_with_bias:
+            input_node = nu.get_node_input(op_node, 0)
+            if input_node.type == 'FakeQuantize':
+                input_node = nu.get_node_input(input_node, 0)
+            calculate_input_shape[input_node.name] = {'shape_node': lambda x: x.shape}
```
Comment (l-bat) [inline tools/pot/openvino/tools/pot/algorithms/quantization/fast_bias_correction/algorithm.py:214]:
```suggestion
            calculate_input_shape[input_node.fullname] = {'shape_node': lambda x: x.shape}
```

File: tools/pot/openvino/tools/pot/algorithms/quantization/fast_bias_correction/algorithm.py [hunk 2]
```diff
@@ -74,7 +75,7 @@ def run(self, model):
                 input_node = nu.get_node_input(input_node, 0)
                 quantized_node = nu.get_node_input(op_node, 0)
 
-            input_shape = nu.get_input_shape_for_bias(op_node)
+            input_shape = inputs_shape_nodes_with_bias[input_node.name]
```
Comment (l-bat) [inline tools/pot/openvino/tools/pot/algorithms/quantization/fast_bias_correction/algorithm.py:78]:
```suggestion
            input_shape = inputs_shape_nodes_with_bias[input_node.fullname]
```

**Comment by KodiaqQ** [comment]
Should we create some examples for this case and add them to the tests? @AlexKoff88, @IndiraSalyahova, what do you think?

File: tools/pot/openvino/tools/pot/algorithms/quantization/fast_bias_correction/algorithm.py [hunk 3]
```diff
@@ -202,3 +203,16 @@ def get_channel_axis(self, node):
         if isinstance(node, tuple):
             return self._channel_axis[node[0]]
         return self._channel_axis[node]
+
+    def get_inputs_shape(self, nodes_with_bias):
```
Comment (KodiaqQ) [inline tools/pot/openvino/tools/pot/algorithms/quantization/fast_bias_correction/algorithm.py:207]: Should we use the same method at the BiasCorrection algorithm, @IndiraSalyahova? It uses the same `get_input_shape_for_bias` for shape calculation.

**Review by AlexKoff88 — APPROVED** [review]

**Pull request merged**